text/0000-release-channels.md

```diff
@@ -0,0 +1,70 @@
+# Bundler release channels
+
+- Feature Name: release\_channels
+- Start Date: 2018-06-30
+- RFC PR:
+- Bundler Issue:
+
+# Summary
+
+Bundler release channels allow users to opt in to the level of release stability that they want. The “canary” channel has a release from every commit that builds green. The default channel has regular releases when features and bugfixes are ready. The “stable” channel only has releases that have fixes for any issues found by the default channel. When possible, bugfixes are backported from the default channel to the stable channel.
+
+# Motivation
+
+As the developers of Bundler, we want to be able to continue to develop and revise Bundler into a better version of itself. As users of Bundler, we want the safest, most stable, least problematic version of Bundler at any given time.
+
+Release channels are a way for the Bundler team to officially bless and support versions that have proven themselves to be stable. It also adds an easy and straightforward way to try the latest upcoming changes to Bundler at any time, increasing the chances that bugs can be found and fixed.
+
+With the stable channel, this RFC aims to actively support the usecases of users deploying Ruby code to important production servers. With the canary channel, this RFC aims to actively support power users, Bundler contributors, early adopters, and anyone who wants to participate in the future of Bundler.
+
+Hopefully, release channels will allow Bundler to change and move forward more easily, while at the same time providing even more stability as an option for users who care most about that.
+
+# Guide-level explanation
+
+To use the default release channel, install the Bundler gem and use it as usual. For example, you install Bundler by running `gem install bundler`, and you install gems by running `bundle install`. The default release of Bundler will let you know anytime there is a newer default release available, but will not install it for you.
+
+To use the stable release channel, instead install with `gem install bundler-stable`. Install gems as usually, by running `bundle install`. The stable version of Bundler will let you know anytime there is a newer stable version available, but will not install it for you. You can identify stable releases of Bundler by running `bundle install --version` and looking for `(stable)` in the output.
```

Comment:
I dont understand how this will work, technically speaking

Reply:
I think one option is shipping a `bundler-stable` gem that overwrites the `bundle` binstub to point to the `bundler-stable` gem instead of `bundler`. (Release channels are global 😛). Maybe another option is to double down on our RubyGems Bundler-loading hackery and prefer the latest stable if it is installed at all?

Reply:
I'm tempted to ask this question: would we be better off adding some support for this into rubygems